File: leetcode/solutions/0283.MoveZeroes/moveZeroes.ts
export default function moveZeroes(nums: number[]): void {
  let insertPos = 0;
    for (const num of nums) {
        if (num !== 0) nums[insertPos++] = num;
    }

    while (insertPos < nums.length) {
        nums[insertPos++] = 0;
    }
}

File: leetcode/solutions/0283.MoveZeroes/moveZeroes.go
package solution

func moveZeroes(nums []int) {
	var i, loopCount int

	for ; loopCount < len(nums); loopCount++ {
		if nums[i] == 0 {
			copy(nums[i:], nums[i+1:])
		} else {
			i++
		}
	}

	zeroCount := loopCount - i
	copy(nums[len(nums)-zeroCount:], make([]int, zeroCount))
}
